lightning-block-sync/src/poll.rs

```diff
@@ -156,16 +158,24 @@ impl std::ops::Deref for ValidatedBlock {
 	}
 }
 
+mod sealed {
+	/// Used to prevent implementing [`super::Validate`] outside the crate but still allow its use.
+	pub trait Validate {}
+
+	impl Validate for crate::BlockHeaderData {}
+	impl Validate for bitcoin::blockdata::block::Block {}
+}
+
 /// The canonical `Poll` implementation used for a single `BlockSource`.
 ///
-/// Other `Poll` implementations must be built using `ChainPoller` as it provides the only means of
-/// validating chain data.
-pub struct ChainPoller<B: DerefMut<Target=T> + Sized , T: BlockSource> {
+/// Other `Poll` implementations should be built using `ChainPoller` as it provides the simplest way
```

Comment:
Doc changes need to be dropped now that its sealed, right?

Reply:
Only `Validate` was sealed. `Poll` can still be implemented.